environment/src/main/java/com/sequenceiq/environment/environment/service/sdx/SdxPollerService.java

```diff
@@ -51,30 +53,37 @@ public class SdxPollerService {
 
     private final WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor;
 
+    private final DatalakeMultipleFlowsResultEvaluator multipleFlowsResultEvaluator;
+
     public SdxPollerService(SdxService sdxService, SdxPollerProvider sdxPollerProvider,
-            WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor) {
+            WebApplicationExceptionMessageExtractor webApplicationExceptionMessageExtractor,
+            DatalakeMultipleFlowsResultEvaluator multipleFlowsResultEvaluator) {
         this.sdxService = sdxService;
         this.sdxPollerProvider = sdxPollerProvider;
         this.webApplicationExceptionMessageExtractor = webApplicationExceptionMessageExtractor;
+        this.multipleFlowsResultEvaluator = multipleFlowsResultEvaluator;
     }
 
     public void startAttachedDatalake(Long envId, String environmentName) {
-        executeSdxOperationAndStartPolling(envId, environmentName, SKIP_START_OPERATION, sdxService::startByCrn, sdxPollerProvider::startSdxClustersPoller);
+        executeSdxOperationAndStartPolling(envId, environmentName, SKIP_START_OPERATION, sdxService::startByCrn, sdxPollerProvider::startStopSdxClustersPoller);
     }
 
     public void stopAttachedDatalakeClusters(Long envId, String environmentName) {
-        executeSdxOperationAndStartPolling(envId, environmentName, SKIP_STOP_OPERATION, sdxService::stopByCrn, sdxPollerProvider::stopSdxClustersPoller);
+        executeSdxOperationAndStartPolling(envId, environmentName, SKIP_STOP_OPERATION, sdxService::stopByCrn, sdxPollerProvider::startStopSdxClustersPoller);
     }
 
     private void executeSdxOperationAndStartPolling(Long envId, String environmentName, Set<SdxClusterStatusResponse> skipStatuses,
-            Consumer<String> sdxOperation, BiFunction<Long, List<String>, AttemptMaker<Void>> attemptMakerFactory) {
+            Function<String, FlowIdentifier> sdxOperation, BiFunction<Long, List<FlowIdentifier>, AttemptMaker<Void>> attemptMakerFactory) {
         try {
-            List<String> sdxCrns = getExecuteSdxOperationsAndGetCrns(environmentName, sdxOperation, skipStatuses);
-            if (CollectionUtils.isNotEmpty(sdxCrns)) {
+            List<FlowIdentifier> flowIdentifiers = getExecuteSdxOperationsAndGetCrns(environmentName, sdxOperation, skipStatuses);
+            if (CollectionUtils.isNotEmpty(flowIdentifiers)) {
                 Polling.stopAfterAttempt(attempt)
-                        .stopIfException(true)
+                        .stopIfException(false)
                         .waitPeriodly(sleeptime, TimeUnit.SECONDS)
-                        .run(attemptMakerFactory.apply(envId, sdxCrns));
+                        .run(attemptMakerFactory.apply(envId, flowIdentifiers));
+            }
+            if (multipleFlowsResultEvaluator.anyFailed(flowIdentifiers)) {
+                throw new EnvironmentServiceException("Sdx operation failed");
```

Comment:
Not sure how complicated it would be to add, but having some logging about which of the polled flows failed would prove helpful. :smile:

Reply:
I will extend the message with the flowIds in this case.